log/log.go

```diff
@@ -6,45 +6,133 @@
 package log
 
 import (
+	"flag"
 	"fmt"
-	golog "log"
+	"log"
+	"log/syslog"
 	"os"
 )
 
 // The following constants represent logging levels in increasing levels of seriousness.
 const (
+	// LevelDebug is the log level for Debug statements.
 	LevelDebug = iota
+	// LevelInfo is the log level for Info statements.
 	LevelInfo
+	// LevelWarning is the log level for Warning statements.
 	LevelWarning
+	// LevelError is the log level for Error statements.
 	LevelError
+	// LevelCritical is the log level for Critical statements.
 	LevelCritical
+	// LevelFatal is the log level for Fatal statements.
 	LevelFatal
 )
 
 var levelPrefix = [...]string{
-	LevelDebug:    "[DEBUG] ",
-	LevelInfo:     "[INFO] ",
-	LevelWarning:  "[WARNING] ",
-	LevelError:    "[ERROR] ",
-	LevelCritical: "[CRITICAL] ",
-	LevelFatal:    "[FATAL] ",
+	LevelDebug:    "DEBUG",
+	LevelInfo:     "INFO",
+	LevelWarning:  "WARNING",
+	LevelError:    "ERROR",
+	LevelCritical: "CRITICAL",
+	LevelFatal:    "FATAL",
 }
 
-// Level stores the current logging level.
-var Level = LevelDebug
+var (
+	// Level stores the current logging level.
+	Level = LevelInfo
+	// sysLogger is a syslog Writer connection.
+	sysLogger *syslog.Writer
+	// UseSyslog controls if syslog is used instead of stdout/err.
+	UseSyslog bool
+	// SyslogNetwork controls the network protocol syslog uses.
+	SyslogNetwork string
+	// SyslogRaddr is the address of a remote syslog instance.
+	SyslogRaddr string
+	// SyslogTag is the tag to pass to syslog.
+	SyslogTag string
+)
 
-func outputf(l int, format string, v []interface{}) {
-	if l >= Level {
-		golog.Printf(fmt.Sprint(levelPrefix[l], format), v...)
+func init() {
+	flag.IntVar(&Level, "loglevel", LevelInfo, "Log level (0 = DEBUG, 5 = FATAL)")
+	flag.BoolVar(&UseSyslog, "syslog", false, "Whether or not to use syslog logging")
+	flag.StringVar(&SyslogNetwork, "syslog-network", "", "Syslog network to use. If left blank along with remote address, will default to local.")
+	flag.StringVar(&SyslogRaddr, "syslog-remote", "", "Syslog server to use. Defaults to local on empty string.")
+	flag.StringVar(&SyslogTag, "syslog-tag", "", "Syslog tag to use. Will use the name of the executable if left blank.")
+}
+
+// SetSyslog sets the default syslog writer (Setting nil will disable syslog).
+func SetSyslog(w *syslog.Writer) {
+	sysLogger = w
+	UseSyslog = sysLogger != nil
+}
+
+func getSysLogger() (*syslog.Writer, error) {
+	if sysLogger != nil {
+		return sysLogger, nil
+	}
+
+	net := SyslogNetwork
+	// This block keeps makes this use local syslog if nothing else is configured.
+	if SyslogRaddr != "" && net == "" {
+		net = "udp"
 	}
+
+	var err error
+	sysLogger, err = syslog.Dial(net, SyslogRaddr, 0, SyslogTag)
+	return sysLogger, err
 }
 
-func output(l int, v []interface{}) {
+func print(l int, msg string) {
 	if l >= Level {
-		golog.Print(levelPrefix[l], fmt.Sprint(v...))
+		msg = fmt.Sprintf("[%s] %s", levelPrefix[l], msg)
```

Comment:
You should only add the levelPrefix when using the Go log library, since it's part of the metadata passed to syslog.